apps/web/components/eventtype/EventAvailabilityTab.tsx

```diff
@@ -278,25 +279,168 @@ const EventTypeSchedule = ({ eventType }: { eventType: EventTypeSetup }) => {
+        ? option.value === scheduleId
+        : availability
+        ? option.value === availability?.value
+        : option.value === schedules.find((schedule) => schedule.isDefault)?.id
+    );
+
+    if (!scheduleId) {
+      setValue(`hosts.${idx}.scheduleId`, value?.value || null, { shouldDirty: true });
+      setValue(`hosts.${idx}.availability`, value || null, { shouldDirty: true });
+    }
+
+    if (!availability) {
+      setValue(`hosts.${idx}.availability`, value || null, { shouldDirty: false });
+    }
+  }, [data]);
+
+  const idx = getIdx();
+  const availabilityValue = watch(`hosts.${idx}.availability`);
+
+  return (
+    <div className="flex w-full flex-col pt-2 ">
+      <Controller
+        name={`hosts.${idx}.scheduleId`}
+        render={({ field }) => {
+          return (
+            <Select
+              placeholder={t("select")}
+              options={options}
+              isSearchable={false}
+              onChange={(selected) => {
+                field.onChange(selected?.value || null);
+                if (selected?.value) {
+                  setValue(`hosts.${idx}.availability`, selected, { shouldDirty: true });
+                }
+              }}
+              className="block w-full min-w-0 flex-1 rounded-sm text-sm"
+              value={availabilityValue}
+              components={{ Option, SingleValue }}
+              isMulti={false}
+              isDisabled={isPending}
+            />
+          );
+        }}
+      />
+    </div>
+  );
+};
+
+export const TeamAvailability = ({ hosts = [] }: { hosts: Host[] }) => {
+  const { t } = useLocale();
+  const [animationRef] = useAutoAnimate<HTMLUListElement>();
+
+  return (
+    <>
+      <div className="border-subtle flex flex-col rounded-md">
+        <div className="border-subtle mt-5 rounded-t-md border p-6 pb-5">
+          <Label className="mb-1 text-sm font-semibold">{t("choose_hosts_schedule")}</Label>
+          <p className="text-subtle max-w-full break-words text-sm leading-tight">
+            {t("hosts_schedule_description")}
+          </p>
+        </div>
+        <div className="border-subtle rounded-b-md border border-t-0 p-6">
+          {hosts.length > 0 ? (
+            <ul
+              className={classNames("mb-4 mt-3 rounded-md", hosts.length >= 1 && "border-subtle border")}
+              ref={animationRef}>
+              {hosts.map((host, index) => (
+                <>
+                  <li
+                    key={host.userId}
+                    className={`flex flex-col px-3 py-2 ${
+                      index === hosts.length - 1 ? "" : "border-subtle border-b"
+                    }`}>
+                    <div className="flex w-full items-center">
+                      <Avatar size="sm" imageSrc={host.avatar} alt={host.label || ""} />
+                      <p className="text-emphasis my-auto ms-3 text-sm">{host.label}</p>
+                    </div>
+                    <TeamMemberSchedule host={host} />
```

Comment:
Pass the index as props instead of having to search it using getIdx.

Reply:
was more cautious , as the hosts index is changed when priority in TeamTab (or Assignment Tab) is changed due to sorting. Since TeamMemberSchedule is child component of TeamAvailability, this is not an issue, tested and updated.